src/lib/spinel/radio_spinel.hpp

```diff
@@ -353,6 +461,44 @@ template <typename InterfaceType> class RadioSpinel
     otError GetCoexMetrics(otRadioCoexMetrics &aCoexMetrics);
 #endif // OPENTHREAD_CONFIG_PLATFORM_RADIO_COEX_ENABLE
 
+    /**
+     * Get currently active interface.
+     *
+     * @param[out] aIid IID of the interface that owns the radio.
+     *
+     * @retval  OT_ERROR_NONE               Successfully got the property.
+     * @retval  OT_ERROR_RESPONSE_TIMEOUT   Failed due to no response received from the transceiver.
+     * @retval  OT_ERROR_NOT_IMPLEMENTED    Failed due to lack of the support in radio or platform supprts
+     *                                      all interfaces simultaneously
+     *                                      (i.e. no active/inactive interface concept in the platform level)
+     *
+     */
+    otError GetMultipanActiveInterface(spinel_iid_t *aIid);
+
+    /**
+     * Sets specified radio interface active
+     *
+     * This function allows selecting currently active radio interface on platforms that do not support parallel
+     * communication on multiple interfaces. I.e. if more than one interface is in receive state calling
+     * SetMultipanActiveInterface guarantees that specified interface will not be losing frames. This function
+     * returns if the request was received properly. After interface switching is complete SwitchoverDone callback is
+     * Invoked. Switching interfaces may take longer if aCompletePending is set true.
+     *
+     * @param[in] aIid              IID of the interface to set active.
+     * @param[in] aCompletePending  Set true if pending radio operation should complete first(Soft switch) or false if
+     * ongoing operations should be interrupted (Force switch).
+     *
+     * @retval  OT_ERROR_NONE               Successfully requested interface switch.
+     * @retval  OT_ERROR_BUSY               Failed due to another operation on going.
+     * @retval  OT_ERROR_RESPONSE_TIMEOUT   Failed due to no response received from the transceiver.
+     * @retval  OT_ERROR_NOT_IMPLEMENTED    Failed due to lack of support in radio for the given interface id or
+     *                                      platform supprts all interfaces simultaneously
+     *                                      (i.e. no active/inactive interface concept in the platform level)
+     * @retval  OT_ERROR_ALREADY            Given interface is already active.
+     *
+     */
+    otError SetMultipanActiveInterface(spinel_iid_t aIid, bool aCompletePending);
```

Comment:
This is probably due to my limited familiarity with the lower level RCP code, but is there an overview of what "parallel communication on multiple interfaces" looks like? Does that typically look like a task popping messages from a queue to see where they are bound, then forwarding them to the appropriate instance? Or is that multiple serial links that are being processed by different tasks? Either?

I think I understand all the technical decisions that are being made here to enable this functionality. However, if I take a step back and think about what "multipan" support for an RCP platform means when its added to the makefile definitions, things are a little more more difficult to understand.

From my naive point of view, "multipan" is any RCP that supports multiple PAN IDs. How that's accomplished internally to the RCP doesn't really matter to someone looking to consume this API. What doc for this function seems to be suggesting is that if an RCP is set up such that the hardware or processing is discrete from each other internally, it is no longer considered "multipan". That seems confusing and potentially hard to reason about during runtime.

I'm more than happy to be incorrect about this, but maybe some documentation around what it means to be "multipan" could be helpful. I don't know if that is in the spec or if that's in the code but I feel like it might be needed if it does not already exist.

Reply:
This is a really good explanation, thank you!

>How multipan is achieved depends on RCP provider

Yeah, I can understand that. I think my confusion was coming from the distinction between full multipan and switching multipan. In my mind, both solutions are multipan from the radio user's perspective, regardless of how the vendor chooses to implement it. But I can see how this function would exclusively be called on switching multipan interfaces.

Does it make sense to document this explanation somewhere? Maybe in the `multipan.h` file? Having this context could be helpful for others.